### DISCUSSION TITLE
Hosting Theia inside a div instead of the body

### DISCUSSION THREAD

**Discussion by nadavsof** [eclipse-theia/theia#9305] [Q&A]
Hi, we wanna host theia wrapped inside our web page rather than the body. There are some parts of the code that directly access document.body so we replaced them and got it working inside a pre-determined div. We ran some (very) basic sanity tests and as far as we can see, , it works fine (more places need to be touched and the unit tests need to be fixed, but it looks doable to us)

A few questions.

1. Are there potential problems that we might be missing, that might occur in edge cases, specific extensions, or future releases, due to this change? Is there a strong reason why theia is currently rendering in the body?
2. Is it on theia's roadmap to add this feature?
3. If it isn't in the roadmap, are you interested in us contributing the change into theia's code base? We can implement it however you'd like, as long as the user can choose to render theia inside a div.

Thanks, Nadav.

**Comment by vince-fugnitto** [comment]
@nadavsof thank you for the discussion!

I was only aware of embedding `theia`-based applications in a larger app through an `iframe`. I have never attempted to wrap it in some parent element, but it seems like changing the default `document.body` is already supported based on the following comment:
- https://github.com/eclipse-theia/theia/issues/1045#issuecomment-376897989
- https://github.com/eclipse-theia/theia/issues/1045#issuecomment-446909231

Based on the discussion it looks like it is already possible for custom `theia`-based applications to customize the behavior instead of using `document.body`. Perhaps this is sufficient for your use-case?
